lib/widgets/amount_form_field.dart

```diff
@@ -24,33 +30,48 @@ class AmountFormField extends TextFormField {
     int maxLength,
     ValueChanged<String> onFieldSubmitted,
     FormFieldSetter<String> onSaved,
-    bool enabled,    
+    bool enabled,
   }) : super(
             focusNode: focusNode,
             keyboardType: TextInputType.numberWithOptions(decimal: true),
-            decoration: decoration,
+            decoration: new InputDecoration(
+              labelText: accountModel.currency.displayName + " Amount",
+              suffixIcon: IconButton(
+                icon: new Image.asset(
+                  accountModel.fiatCurrency.logoPath,
+                  color: iconColor != null ? iconColor : theme.BreezColors.white[500],
+                ),
+                padding: EdgeInsets.only(top: 21.0),
+                alignment: Alignment.bottomRight,
+                onPressed: () => showDialog(
+                    context: context,
+                    builder: (_) => CurrencyConverterDialog(returnFN != null ? returnFN : (value) => controller.text = value)),
+              ),
+            ),
             style: style,
             enabled: enabled,
             controller: controller,
-            inputFormatters: currency != Currency.SAT ? [WhitelistingTextInputFormatter(RegExp(r'\d+\.?\d*'))] : [WhitelistingTextInputFormatter.digitsOnly],
+            inputFormatters: accountModel.currency != Currency.SAT
+                ? [WhitelistingTextInputFormatter(RegExp(r'\d+\.?\d*'))]
+                : [WhitelistingTextInputFormatter.digitsOnly],
             onFieldSubmitted: onFieldSubmitted,
             onSaved: onSaved);
 
   @override
   FormFieldValidator<String> get validator {
     return (value) {
       if (value.isEmpty) {
-        return "Please enter the amount in " + currency.displayName;
+        return "Please enter the amount in " + accountModel.currency.displayName;
       }
-      Int64 intAmount = currency.parse(value);
+      Int64 intAmount = accountModel.currency.parse(value);
       if (intAmount <= 0) {
         return "Invalid amount";
       }
       String msg;
-      if (validatorFn != null) {
-        msg = validatorFn(intAmount);
-      }   
-      return msg;   
+      if (accountModel.validateOutgoingOnChainPayment != null) {
```

Comment:
As part of simplifying the use of AmountFormField the validator now always use `accountModel.validateOutgoingOnChainPayment` where on some cases it should use `accountModel. validateOutgoingPayment` or `accountModel.validateIncomingPayment`. I think for now it is better to pass this function as parameter (as was before the change).